index.js

```diff
@@ -36,17 +36,9 @@ function getKeys(target) {
 	return Object.keys(target).concat(getEnumerableOwnPropertySymbols(target))
 }
 
-function propertyIsOnObject(object, property) {
-	try {
-		return property in object
-	} catch(_) {
```

Comment:
hmm, in what case was `in` throwing before?  If there's no unit test for it, I wonder if we can find evidence in the commit log or old issues

Reply:
See the benchmark, which is based on the unit test "merge nested objects into target"

If key1 of the target is a string, then you cant add sub attributes to it, as it is not an object. In that case you are overwriting the target with the value of the src.

Reply:
@TehShrike does this answer your concerns/questions?